src/crimson/osd/osd_operation.cc

```diff
@@ -75,11 +75,11 @@ void OSDOperationRegistry::put_historic(const ClientRequest& op)
   // NOTE: Operation uses the auto-unlink feature of boost::intrusive.
   // NOTE: the cleaning happens in OSDOperationRegistry::do_stop()
   using crimson::common::local_conf;
-  if (num_recent_ops >= local_conf()->osd_op_history_size) {
-    ++last_of_recents;
+  if (get_duration(op) > local_conf()->osd_op_complaint_time) {
     ++num_slow_ops;
   } else {
     ++num_recent_ops;
+    --last_of_recents;
```

Comment:
I'm confused as to what last_of_recents is supposed to indicate.  Is it the case that for all ops in the interval [last_of_recents, std::end(historic_registry)), get_duration(op) <= local_conf()->osd_op_complaint_time?  If so, I'm not sure this patch actually maintains that invariant.